source/adios2/toolkit/format/bp5/BP5Deserializer.cpp

```diff
@@ -1219,6 +1219,7 @@ bool BP5Deserializer::QueueGetSingle(core::VariableBase &variable,
             Req.Count = variable.m_Count;
         }
         Req.Data = DestData;
+        Req.MemSpace = MemSpace;
```

Comment:
I don't know how I missed this :) I guess the testing I added only looks at one particular example. I double checked and I think this is the only place missing the memory space

Reply:
> I don't know how I missed this :) I guess the testing I added only looks at one particular example. I double checked and I think this is the only place missing the memory space

Unfortunately we run msan only on some smallish subset of our test suite.  Partly because MPI is notoriously bad for memory abuse (so no MPI runs) and partly because msan really slows down tests and we can't have 10 hour CI runs (so some tests get eliminated).  As a result, uninit fields may slip by sometimes...